assets/js/atomic/blocks/product-elements/rating/edit.tsx

```diff
@@ -33,11 +34,25 @@ const Edit = ( {
 		...context,
 	};
 	const isDescendentOfQueryLoop = Number.isFinite( context.queryId );
+	const { isDescendentOfSingleProductBlock } = useSelect( ( select ) => {
```

Comment:
Would it be worth having this logic stored as a React hook for shared usage across all inner blocks for the Single Product? Could other blocks, such as price and add to cart, also benefit from this change?

Reply:
Yes! I agree with this, today I opened another PR to an issue that I was investigating and I found out that it was being caused by the same thing, so I created a hook [in that PR](https://github.com/woocommerce/woocommerce-blocks/pull/9446) that I'll be using to refactor this and other blocks that need it